File: Java_programming/check_prime.py
# write a program which accepts one number and checks if it prime or not

def ChkPrime(iNo):
    if(iNo <= 1):
        return False

    bFlag = True

    for i in range(2,iNo//2 + 1):
        if(iNo % i == 0):
            bFlag = False
            break
    
    return bFlag

def main():

    iValue = 0
    bRet = True

    print("Enter the number : ",end="")
    iValue = int(input())

    bRet = ChkPrime(iValue)

    if(bRet == False):
        print(iValue,"is not a prime number")
    else:
        print(iValue,"is a prime number")

if __name__ == "__main__":
    main()

File: Cpp_programming/numbers/check_prime.cpp
// write a program which accpets one number and check whether it prime or not
// I/P :- 11
// O/P :- true

#include<iostream>
using namespace std;

bool CheckPrime(int iNo)
{
    bool bFlag = true;
    if(iNo <= 1)
    {
        return false;
    }

    int i = 0;
    for(i = 2; i <= iNo/2; i++)
    {
        if(iNo % i == 0)
        {
            bFlag = false;
        }
    }
    return bFlag;
}

int main()
{
    int iVal = 0;
    bool bRet = false;

    cout<<"Enter the number :- ";
    cin>>iVal;

    bRet = CheckPrime(iVal);

    if(bRet == true)
    {
        cout<<iVal<<" is prime number\n";
    }
    else
    {
        cout<<iVal<<" is not prime number\n";
    }
    return 0;
}